src/freenet/clients/fcp/ClientRequest.java

```diff
@@ -27,6 +27,9 @@
  */
 public abstract class ClientRequest implements Serializable {
 
+    /**
+     * ATTENTION: When incrementing this, please skip version 2. Version 2 had already temporarily
+     * been used by a development branch. */
```

Comment:
I had initially implemented this like you demand here: The initial version bumped the UID to 2, so the serialization code would not eat the files; and then fallback to the backup which is stored in a different format. (Nice that toad implemented this!)

This would have however caused loss of uploads since the backup code is only implemented for downloads.
I did not care about that since we have no reason to support compatibility for testing releases as you said.

But @ArneBab rightfully argued that purge-db4o beta versions have been out for a very long time and many people might be using them.

I would suggest the following solution:
- I refrain from casting a vote here since I already had a heavy discussion with @ArneBab  and I think he won. You instead discuss with @ArneBab  whether to keep as is or drop compatibility.
- To avoid such issues in the future, make a decision whether testing releases may causs loss of downloads/etc. Also, always warn people about potential loss of downloads/uploads with testing releases in the release notes.